content/automate/ManageIQ/Cloud/VM/Provisioning/Placement.class/__methods__/best_fit_azure.rb

```diff
@@ -1,37 +1,66 @@
+#              and resource group for Azure                       #
+#                                                                 #
 ###################################################################
+module ManageIQ
+  module Automate
+    module Cloud
+      module VM
+        module Provisioning
+          module Placement
+            class BestFitAzure
+              def initialize(handle = $evm)
+                @handle = handle
+              end
 
-$evm.log("info", "Using Auto Placement for Azure Cloud Provider")
-prov  = $evm.root["miq_provision"]
-image = prov.vm_template
-raise "Image not specified" if image.nil?
+              def main
+                @handle.log("info", "Using Auto Placement for Azure Cloud Provider")
+                @prov = @handle.root["miq_provision"]
+                raise "Image not specified" if @prov.try(:vm_template).nil?
 
-if prov.get_option(:cloud_network).nil?
-  cloud_network = prov.eligible_cloud_networks.first
+                @prov.get_option(:cloud_network)  || default_cloud_network
+                @prov.get_option(:cloud_subnet)   || default_cloud_subnet
+                @prov.get_option(:resource_group) || default_resource_group
+              end
 
-  if cloud_network
-    prov.set_cloud_network(cloud_network)
-    $evm.log("info", "Selected Cloud Network: #{cloud_network.name}")
-  end
-end
+              private
 
-if prov.get_option(:cloud_subnet).nil?
-  cloud_subnet = prov.eligible_cloud_subnets.first
+              def default_cloud_network
+                cloud_network = @prov.eligible_cloud_networks.first
 
-  if cloud_subnet
-    prov.set_cloud_subnet(cloud_subnet)
-    $evm.log("info", "Selected Cloud Subnet: #{cloud_subnet.name}")
-  end
-end
+                if cloud_network
+                  @prov.set_cloud_network(cloud_network)
+                  @handle.log("info", "Selected Cloud Network: #{cloud_network.name}")
+                end
+              end
 
-if prov.get_option(:resource_group).nil?
-  resource_group = prov.eligible_resource_groups.first
+              def default_cloud_subnet
+                cloud_subnet = @prov.eligible_cloud_subnets.first
 
-  if resource_group
-    prov.set_resource_group(resource_group)
-    $evm.log("info", "Selected Resource Group: #{resource_group.name}")
+                if cloud_subnet
+                  @prov.set_cloud_subnet(cloud_subnet)
+                  @handle.log("info", "Selected Cloud Subnet: #{cloud_subnet.name}")
+                end
+              end
+
+              def default_resource_group
+                resource_group = @prov.eligible_resource_groups.first
+
+                if resource_group
+                  @prov.set_resource_group(resource_group)
+                  @handle.log("info", "Selected Resource Group: #{resource_group.name}")
+                end
+              end
+            end
+          end
+        end
+      end
+    end
   end
 end
+
+if $PROGRAM_NAME == __FILE__
```

Comment:
@pkomanek 
Based on this PR https://github.com/ManageIQ/manageiq-content/pull/390 we dont need the if block anymore you only need line number 65 by itself and the spec_helper will mock this class at load time